torch/csrc/deploy/interpreter/interpreter_impl.cpp

```diff
@@ -0,0 +1,455 @@
+  int res = 0;
+
+  /* Count the number of entries in both tables */
+  for (a = 0; frozenpython[a].name != nullptr; a++) {
+    // std::cout << "frozenpython[" << a << "]: " << frozenpython[a].name <<
+    // std::endl;
+  }
+  for (b = 0; frozentorch[b].name != nullptr; b++) {
+    // std::cout << "frozentorch[" << b << "]: " << frozentorch[b].name <<
+    // std::endl;
+  }
+  for (c = 0; PyImport_FrozenModules[c].name != nullptr; c++) {
+    // std::cout << "oldfrozen[" << c << "]: " << PyImport_FrozenModules[c].name
+    // << std::endl;
+  }
+
+  // Num frozen builtins shouldn't change (unless modifying the underlying
+  // cpython version)
+  TORCH_INTERNAL_ASSERT(
+      c == NUM_FROZEN_PY_BUILTIN_MODULES,
+      "Missing python builtin frozen modules");
+  // Check a+b together since in OSS a is empty and b contains stdlib+torch,
+  // while in fbcode they are separated due to thirdparty2 frozenpython. No
+  // fixed number of torch modules to check for, but there should be at least
+  // one.
+  TORCH_INTERNAL_ASSERT(
+      a + b > NUM_FROZEN_PY_STDLIB_MODULES + 1,
+      "Missing frozen python stdlib or torch modules");
+
+  /* Allocate new memory for the combined table */
+  if (a + b + c <= SIZE_MAX / sizeof(struct _frozen) - 1) {
+    size_t size = sizeof(struct _frozen) * (a + b + c + 1);
+    p = (_frozen*)PyMem_Realloc(p, size);
+  }
+  if (p == nullptr) {
+    return -1;
+  }
+
+  /* Copy the tables into the new memory */
+  memcpy(p, PyImport_FrozenModules, (c + 1) * sizeof(struct _frozen));
+  memcpy(p + c, frozenpython, (a + 1) * sizeof(struct _frozen));
+  memcpy(p + a + c, frozentorch, (b + 1) * sizeof(struct _frozen));
+  PyImport_FrozenModules = p;
+  return res;
+}
+
+static py::object global_impl(const char* module, const char* name) {
+  return py::module::import(module).attr(name);
+}
+
+using at::IValue;
+using torch::PickledObject;
+using torch::PythonObject;
+
+struct ScopedAcquire {
+  ScopedAcquire() {
+    PyGILState_Ensure();
+  }
+  ~ScopedAcquire() {
+    PyEval_SaveThread();
+  }
+};
+
+struct InitLockAcquire {
+  InitLockAcquire(std::mutex& init_lock) : init_lock_(init_lock) {
+    // to avoid deadlock, we need to ensure a consistent lock order:
+    // init_lock -> GIL. Otherwise, the GIL can be released by the python
+    // interpreter during initalization tasks, and then re-acquired. If another
+    // thread grabs the GIL to do non-initialization tasks, then it might start
+    // initializing (GIL -> init_lock). To avoid this, releasethe GIL before
+    // trying to get the init_lock and then reacquire it afterward.
+    PyEval_SaveThread();
+    init_lock.lock();
+    PyGILState_Ensure();
+  }
+  ~InitLockAcquire() {
+    init_lock_.unlock();
+  }
+
+ private:
+  std::mutex& init_lock_;
+};
+
+struct ConcreteInterpreterImpl : public torch::InterpreterImpl {
+  ConcreteInterpreterImpl() {
+    // some dependency in mkl requires this...
+    void* result = dlopen("libz.so", RTLD_GLOBAL | RTLD_LAZY);
```

Comment:
is it a recent thing in OSS build or some earlier workaround for fbcode. There was a fix recently for something libz related (https://fburl.com/bibq1q9z)